kotlin/src/main/kotlin/net/jqwik/kotlin/api/JqwikGlobals.kt

```diff
@@ -106,6 +106,70 @@ inline fun <reified T> anyForType(): TypeArbitrary<T>
     return Arbitraries.forType(T::class.java)
 }
 
+
+/**
+ * Creates [Arbitrary] with subtypes of a sealed class or interface [T].
+ * If a subtype is a sealed class or interface, its subtypes are used to create [Arbitrary]. This is done recursively.
+ * [TypeArbitrary] are created by default under the hood, but this can be customized, for each subtype, with [SubtypeScope.provide] .
+ * ```kotlin
+ * anyForSubtypeOf<MyInterface> {
+ *     provide<MyImplementation1> { customArbitrary1() }
+ *     provide<MyImplementation2> { customArbitrary2() }
+ * }
+ * ```
+ * @param enableArbitraryRecursion is applied to all created [TypeArbitrary].
+ */
+inline fun <reified T> anyForSubtypeOf(
+    enableArbitraryRecursion: Boolean = false,
+    crossinline subtypeScope: SubtypeScope<T>.() -> Unit = {}
+): Arbitrary<T> where T : Any {
```

Comment:
Would it be possible to have `TypeArbitrary<T>` as return type?

Reply:
This would imply that custom arbitrary (registered with `SubtypeScope<T>.provide()`) can only be `TypeArbitrary<T>`. This would make custom arbitraries useless, wouldn't it?